src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private static final int MAX_SIZE = 10;
+    private static final int DEFAULT_VALUE = 10;
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int values = random.nextInt(NUMBER_OF_FIGURES);
+        switch (values) {
+            case 0:
+                return new Square(new ColorSupplier().getRandomColor(), random.nextInt(MAX_SIZE));
+            case 1:
+                return new Rectangle(new ColorSupplier().getRandomColor(),
+                        random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE));
+            case 2:
+                return new RightTriangle(new ColorSupplier().getRandomColor(),
+                        random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE));
+            case 3:
+                return new IsoscelesTrapezoid(new ColorSupplier().getRandomColor(),
+                        random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE));
+            default:
+                return new Circle(new ColorSupplier().getRandomColor(), random.nextInt(MAX_SIZE));
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(String.valueOf(Color.WHITE), DEFAULT_VALUE);

```

Comment:
Don’t use toString() or String.valueOf()(it will call toString() under the hood) for getting String representation of enum constants. toString() is common for all enum constants. So it’s better to use standard method of enum name() that will be returning always String representation of concrete enum constant.